pkg/tenant/members.go

```diff
@@ -261,6 +262,105 @@ func (s *Server) MemberUpdate(c *gin.Context) {
 	c.JSON(http.StatusOK, m.ToAPI())
 }
 
+func (s *Server) MemberRoleUpdate(c *gin.Context) {
+	var err error
+
+	// Members exist on organizations
+	// This method handles the logging and error responses
+	var orgID ulid.ULID
+	if orgID = orgIDFromContext(c); ulids.IsZero(orgID) {
+		return
+	}
+
+	// Get the member ID from the URL and return a 400 if the member ID is not a ULID.
+	var memberID ulid.ULID
+	if memberID, err = ulid.Parse(c.Param("memberID")); err != nil {
+		sentry.Warn(c).Err(err).Str("id", c.Param("memberID")).Msg("could not parse member id")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("member not found"))
+		return
+	}
+
+	// TODO: Add org verification
+
+	// Bind the user request with JSON.
+	params := &api.UpdateMemberParams{}
+	if err = c.BindJSON(&params); err != nil {
+		sentry.Warn(c).Err(err).Msg("could not parse member update request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrUnparsable))
+		return
+	}
+
+	// Verify member role exists.
+	if params.Role == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("member role is required"))
+		return
+	}
+
+	// Verify the role provided is valid.
+	perms.IsRole(params.Role)
+
+	// Get members from the database and set page size to return all members.
+	// TODO: Create list method that will not require pagination for this endpoint.
+	getAll := &pg.Cursor{StartIndex: "", EndIndex: "", PageSize: 100}
+	var members []*db.Member
+	if members, _, err = db.ListMembers(c.Request.Context(), orgID, getAll); err != nil {
+		sentry.Error(c).Err(err).Msg("could not list members")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not update member role"))
+		return
+	}
+
+	// Loop over dbMember and count the number of members whose role is Owner to verify that at least one Owner remains in the organization.
+	var count bool
+	for _, dbMember := range members {
+		if dbMember.Role == perms.RoleOwner {
+			count = true
+			break
+		}
+	}
```

Comment:
It's possible we will want to put this in a helper method, since I think we will also want to check this on MemberDelete (to prevent the only Owner in an organization from being deleted). Can we put a TODO here to note that for now?